src/ralph/admin/views.py

```diff
@@ -104,3 +107,30 @@ def get_url_pattern(cls, model):
         return r'^{}/{}/(?P<pk>[0-9]+)/{}/$'.format(
             model._meta.app_label, model._meta.model_name, cls.url_name
         )
+
+
+class WithAdminBase(type):
+    def __new__(cls, name, bases, attrs):
+        empty_fieldset = (('__empty__', {'fields': []}),)
+        admin_whitelist = ['inlines', 'fieldsets']
+        admin_attrs = {
+            key: attrs.pop(key, None) for key in admin_whitelist
+        }
+        admin_attrs['change_form_template'] = attrs.get('template_name')
+        admin_attrs['fieldsets'] = admin_attrs['fieldsets'] or empty_fieldset
+        new_class = type.__new__(cls, name, bases, attrs)
+        new_class.admin_class = type('AdminView', (RalphAdmin,), admin_attrs)
+        return new_class
+
+
+class WithAdmin(metaclass=WithAdminBase):
```

Comment:
check if this can inherit from `RalphDetailView` (and then `NetworkView` could inherit only from one class)